library/modules/EventManager.cpp

```diff
@@ -53,11 +58,29 @@ using namespace df::enums;
  *  consider a typedef instead of a struct for EventHandler
  **/
 
+static void write_persistent(color_ostream& out);
+static bool already_wrote_to_file = false;
+struct nemesis_hook : df::general_ref_nemesis {
+    typedef df::general_ref_nemesis interpose_base;
+    DEFINE_VMETHOD_INTERPOSE(void,write_file,(df::file_compressorst* file)) {
+        CoreSuspendClaimer suspend;
+        color_ostream_proxy out(Core::getInstance().getConsole());
+        //out << "write_file" << endl;
+        if ( !already_wrote_to_file ) {
+            already_wrote_to_file = true;
+            //out << "write_persistent" << endl;
+            write_persistent(out);
+        }
+        INTERPOSE_NEXT(write_file)(file);
+    }
+};
+IMPLEMENT_VMETHOD_INTERPOSE(nemesis_hook, write_file);
+
```

Comment:
Clever.  It looks like this will write the file on seasonal autosaves too, right?  I'm less sure about whether it writes when saving the world after a paused autosave, in case configuration has been changed without unpausing first, but manageEvents should get called in between somewhere, right?